protocol/communities/manager.go

```diff
@@ -1478,107 +1480,90 @@ func (m *Manager) HandleCommunityEventsMessageRejected(signer *ecdsa.PublicKey,
+			State:       state,
 		}
+		requestToJoin.CalculateID()
 
-		var state RequestToJoinState
-		if message.Type == protobuf.CommunityPrivilegedUserSyncMessage_CONTROL_NODE_ACCEPT_REQUEST_TO_JOIN {
-			state = RequestToJoinStateAccepted
-		} else {
-			state = RequestToJoinStateDeclined
+		_, err := m.saveOrUpdateRequestToJoin(signer, communityID, requestToJoin)
+		if err != nil {
+			return nil, err
 		}
+		requestsToJoin = append(requestsToJoin, requestToJoin)
+	}
 
-		for signer, requestToJoinProto := range message.RequestToJoin {
-			requestToJoin := &RequestToJoin{
-				PublicKey:   signer,
-				Clock:       requestToJoinProto.Clock,
-				ENSName:     requestToJoinProto.EnsName,
-				CommunityID: requestToJoinProto.CommunityId,
-				State:       state,
-			}
-			requestToJoin.CalculateID()
+	return requestsToJoin, nil
+}
 
-			_, err := m.saveOrUpdateRequestToJoin(signer, community.ID(), requestToJoin)
-			if err != nil {
-				return err
-			}
+func (m *Manager) HandleAddCommunityTokenPrivilegedUserSyncMessage(message *protobuf.CommunityPrivilegedUserSyncMessage, community *Community) error {
+	for _, token := range message.CommunityTokens {
+		token := community_token.FromCommunityTokenProtobuf(token)
+		if !community.MemberCanManageToken(community.MemberIdentity(), token) {
+			return ErrInvalidManageTokensPermission
 		}
-	case protobuf.CommunityPrivilegedUserSyncMessage_ADD_COMMUNITY_TOKENS:
-		for _, token := range message.CommunityTokens {
-			token := community_token.FromCommunityTokenProtobuf(token)
-			if !community.MemberCanManageToken(community.MemberIdentity(), token) {
-				return ErrInvalidManageTokensPermission
-			}
 
-			exist, err := m.persistence.HasCommunityToken(token.CommunityID, token.Address, token.ChainID)
-			if err != nil {
-				return err
-			}
-
-			if !exist {
-				return m.persistence.AddCommunityToken(token)
-			}
+		exist, err := m.persistence.HasCommunityToken(token.CommunityID, token.Address, token.ChainID)
+		if err != nil {
+			return err
 		}
 
+		if !exist {
+			return m.persistence.AddCommunityToken(token)
+		}
 	}
-
 	return nil
 }
 
-func (m *Manager) handleAdditionalAdminChanges(community *Community) error {
+func (m *Manager) handleAdditionalAdminChanges(community *Community) (*CommunityResponse, error) {
 
 	if !(community.IsControlNode() || community.HasPermissionToSendCommunityEvents()) {
 		// we're a normal user/member node, so there's nothing for us to do here
-		return nil
+		return nil, nil
+	}
+
+	communityResponse := CommunityResponse{
+		RequestsToJoin: make([]*RequestToJoin, 0),
 	}
 
 	for i := range community.config.EventsData.Events {
 		communityEvent := &community.config.EventsData.Events[i]
 		switch communityEvent.Type {
 		case protobuf.CommunityEvent_COMMUNITY_REQUEST_TO_JOIN_ACCEPT:
-			err := m.handleCommunityEventRequestAccepted(community, communityEvent)
+			requestsToJoin, err := m.handleCommunityEventRequestAccepted(community, communityEvent)
```

Comment:
Similarly here, when handling additional admin changes, I return the requests to join so that I can update the activity center in the messenger.